### PR TITLE
Module reorganization and fixes to Travis

### PR DESCRIPTION
**Pull request by jpkrohling**

### CODE REVIEW

**Comment by jpkrohling** [comment]
@pinpong , this is just an initial draft for the three next steps from #32 . I'd appreciate your review once you have some time. No actual code should have been changed, only reorganizing the modules. Once it's reviewed and merge, I'll send my PR for the `Kerl.java` and `Converter`

**Review by adrianziser — APPROVED** [review]

**Review by adrianziser — CHANGES_REQUESTED** [review]
Oli has requested alon to review this but i dont think he will find the time atm. i just had a look and it looks good for me + i welcome the changes. Sorry that i merged other pr before yours, could you please resolve the conflicts and resubmit your pr that we can finally accept & merge it?

**Comment by jpkrohling** [comment]
Sorry for the delay, I've been a bit busy for the last couple of weeks but I'll rebase and update this PR soon.

**Comment by jpkrohling** [comment]
This PR has been rebased and updated to fix the merge conflicts.

File: pom.xml [hunk 1]
```diff
@@ -1,145 +1,137 @@
...
-            <groupId>commons-io</groupId>
-            <artifactId>commons-io</artifactId>
-            <version>2.5</version>
-        </dependency>
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.bouncycastle</groupId>
+                <artifactId>bcprov-jdk15on</artifactId>
+                <version>1.57</version>
```
Comment (th0br0) [inline pom.xml]: Could you bump to `1.58`? There were a number of Keccak(Kerl) performance improvements.

**Comment by th0br0** [comment]
Looks sane to me! Could you just bump the bouncycastle version?

**Comment by jpkrohling** [comment]
Version bumped

**Pull request merged**